frontend_tests/node_tests/message_list.js

```diff
@@ -234,3 +234,38 @@ var MessageList = require('js/message_list').MessageList;
         });
     });
 }());
+
+(function test_unmuted_messages() {
+    var table;
+    var filter = {};
+
+    var list = new MessageList(table, filter);
+
+    var unmuted = [
+        {
+            id: 50,
+            stream: 'bad',
+            mentioned: true,
+        },
+        {
+            id: 60,
+            stream: 'good',
+            mentioned: false,
+        },
+    ];
+    var muted = [
+        {
+            id: 70,
+            stream: 'bad',
+            mentioned: false,
+        },
+    ];
+
+    with_overrides(function (override) {
+        override('muting.is_topic_muted', function (stream) {
+            return stream === 'bad';
+        });
+        var test_unmuted = list.unmuted_messages(unmuted.concat(muted));
+        assert.deepEqual(unmuted, test_unmuted);
+    });
```

Comment:
A common explaining the purpose of this test might be useful.  Could be a good idea for some of the later tests as well.